src/lib/lib_duplicity.py

```diff
@@ -166,10 +172,16 @@ def _render_backup_cron(self):
         backup_frequency = self.charm_config.get("backup_frequency")
         if backup_frequency in ["hourly", "daily", "weekly", "monthly"]:
             backup_frequency = "@{}".format(backup_frequency)
+        is_juju3 = subprocess.getoutput("ls /usr/bin/ | grep juju-exec")
+        if is_juju3:
+            juju_binary = "juju-exec"
+        else:
+            juju_binary = "juju-run"
         cron_ctx = {
             "frequency": backup_frequency,
             "unit_name": hookenv.local_unit(),
             "charm_dir": hookenv.charm_dir(),
+            "juju_command": juju_binary,
```

Comment:
**nitpick:** 

Why don't use the same name as `juju_exec_binary`

Reply:
**nitpick**: maybe consider making it a short private function, and add unit test for that.

e.g.
```python
def _get_juju_exec_binary(self):
    ...
```

Reply:
sure will do it in another PR.